tensornetwork/block_sparse/charge.py

```diff
@@ -306,35 +305,56 @@ def intersect(self, other, assume_unique=False, return_indices=False) -> Any:
     if return_indices:
       obj.__init__(
           charges=out[0],
-          charge_labels=np.arange(len(out[0]), dtype=np.int16),
+          charge_labels=np.arange(out[0].shape[1], dtype=np.int16),
           charge_types=self.charge_types,
       )
       return obj, out[1], out[2]
 
     obj.__init__(
         charges=out,
-        charge_labels=np.arange(len(out), dtype=np.int16),
+        charge_labels=np.arange(out.shape[1], dtype=np.int16),
         charge_types=self.charge_types,
     )
 
     return obj
 
+  def sort_unique_charges(self) -> "BaseCharge":
+    """
+    Sort the `unique_charges` of BaseCharge` according to standard order 
+    used by numpy.
+    Returns:
+      BaseCharge
+    """
+    unique_charges, inverse = np.unique(
+        self.unique_charges, return_inverse=True, axis=1)
+    charge_labels = inverse[self.charge_labels]
+    obj = self.__new__(type(self))
+    obj.__init__(
+        charges=unique_charges,
+        charge_labels=charge_labels,
+        charge_types=self.charge_types)
+    return obj
+
   def unique(self,
-             return_index=False,
-             return_inverse=False,
-             return_counts=False) -> Any:
+             return_index: Optional[bool] = False,
+             return_inverse: Optional[bool] = False,
+             return_counts: Optional[bool] = False,
+             sort: Optional[bool] = True) -> Any:
```

Comment:
Optional means these arguments can be None, is that what you actually want here?

Reply:
Oh, actually no. They shoud be `bool`

Reply:
removed the `Optional`.

Reply:
turns out that `None` is fine, put `Optional` back

Reply:
There's a difference between fine and correct. Let's keep it as just bool.